File: Validator.cpp
#include "Validator.h"
#include <iostream>
#include <sstream>

int Validator::validateAndGetInt(const std::string &prompt) {
  std::string input;
  int value;

  do {
    std::cout << prompt;
    std::cin >> input;

    std::istringstream iss(input);
    if (iss >> value) {
      // Check if the value is non-negative
      if (value >= 0) {
        return value;
      } else {
        std::cout << "Please enter a non-negative value.\n";
      }
    } else {
      std::cout << "Invalid input. Please enter a valid number.\n";
    }
  } while (true);
}

File: validator.py
# validator.py


class Validator:
    @staticmethod
    def validate_numeric_input(prompt):
        while True:
            print(prompt)
            user_input = input()
            try:
                value = float(user_input)
                if value < 0:
                    raise ValueError("Please enter a non-negative value.")
                return value
            except ValueError:
                print("Invalid input. Please enter a valid non-negative number.")
                user_input = input()
